inc/user.class.php

```diff
@@ -675,54 +675,47 @@ function prepareInputForUpdate($input) {
          self::dropPictureFiles($this->fields['picture']);
          $input['picture'] = 'NULL';
       } else {
-
-         if (isset($_FILES['picture'])) {
-            if (!count($_FILES['picture'])
-                || empty($_FILES['picture']['name'])
-                || !is_file($_FILES['picture']['tmp_name'])) {
-
-               switch ($_FILES['picture']['error']) {
-                  case UPLOAD_ERR_INI_SIZE :
-                  case UPLOAD_ERR_FORM_SIZE :
-                     Session::addMessageAfterRedirect(__('File too large to be added.'), false,
-                                                      ERROR);
-                     break;
-
-                  case UPLOAD_ERR_NO_FILE :
-                     // Session::addMessageAfterRedirect(__('No file specified.'),false,ERROR);
-                     break;
-               }
-
-            } else {
-               if (toolbox::getMime($_FILES['picture']['tmp_name'], 'image')) {
-                  // Unlink old picture (clean on changing format)
-                  self::dropPictureFiles($this->fields['picture']);
-                  // Move uploaded file
-                  $filename     = uniqid($this->fields['id'].'_');
-                  $sub          = substr($filename, -2); /* 2 hex digit */
-                  $tmp          = explode(".", $_FILES['picture']['name']);
-                  $extension    = array_pop($tmp);
-                  @mkdir(GLPI_PICTURE_DIR . "/$sub");
-                  $picture_path = GLPI_PICTURE_DIR  . "/$sub/${filename}.$extension";
-                  self::dropPictureFiles($filename.".".$extension);
-
-                  if (in_array($extension, ['jpg', 'jpeg', 'png', 'bmp', 'gif'])
-                      && Document::renameForce($_FILES['picture']['tmp_name'], $picture_path)) {
-                     Session::addMessageAfterRedirect(__('The file is valid. Upload is successful.'));
-                     // For display
-                     $input['picture'] = "$sub/${filename}.$extension";
-
-                     //prepare a thumbnail
-                     $thumb_path = GLPI_PICTURE_DIR . "/$sub/${filename}_min.$extension";
-                      Toolbox::resizePicture($picture_path, $thumb_path);
-                  } else {
-                     Session::addMessageAfterRedirect(__('Potential upload attack or file too large. Moving temporary file failed.'),
-                                                     false, ERROR);
-                  }
+         $newPicture = false;
+         if (isAPI()) {
```

Comment:
i think this whole part (if...else) could be factorized a little.
It seems actually a little redondant.